Doc updates (#159)
open-ce/open-ce: doc/README.build_env.md

                        False)
==============================================================================
```

## Conda environment files

> `build_env.py` also generates conda environments file based on the configuration

Reviewer: ```suggestion
`build_env.py` also generates conda environment files based on the configuration
```

> `build_env.py` also generates a conda environment file based on the configuration

Reviewer: ```suggestion
`build_env.py` also generates conda environments file based on the configuration
```

>  selected for build. For e.g. if `build_env.py` is run for tensorflow-env.yaml and

Reviewer 2: selected for build. For e.g. if `build_env.py` is run for `tensorflow-env.yaml` and

>  selected for build. For e.g. if `build_env.py` is run for tensorflow-env.yaml and

Reviewer 2: ```suggestion
 selected for build. For e.g. if `build_env.py` is run for `tensorflow-env.yaml` and
```

>  selected for build. For e.g. if `build_env.py` is run for `tensorflow-env.yaml` and

Reviewer: ```suggestion
 selected for a build. For e.g. if `build_env.py` is run for `tensorflow-env.yaml` and
```

>  for python version 3.7, build_type `cuda` and mpi_type being openmpi, then a

Reviewer 2: ```suggestion
 for python version `3.7`, build_type `cuda` and mpi_type being `openmpi`, then a
```

>  conda environment file with name open-ce-conda-env-py3.7-cuda-openmpi.yaml gets

Reviewer 2: ```suggestion
 conda environment file with name `open-ce-conda-env-py3.7-cuda-openmpi.yaml` gets
```

 generated. This environment file can be used to create a conda environment with

>  the packages listed in tensorflow-env.yaml installed in it.

Reviewer 2: ```suggestion
 the packages listed in `tensorflow-env.yaml` installed in it.
```

```shell
    ./open-ce/build_env.py --python_versions=3.7 --build_type=cuda --mpi_type=openmpi
    envs/tensorflow-env.yaml
```

>  The above command will output open-ce-conda-env-py3.7-cuda-openmpi.yaml in the specified

Reviewer 2: ```suggestion
 The above command will output `open-ce-conda-env-py3.7-cuda-openmpi.yaml` in the specified
```

 output folder (or by default `./condabuild` directory).

 The following command can be used to create a conda environment using the generated conda
 environment file -

```shell
    conda env create -f open-ce-conda-env-py3.7-cuda-openmpi.yaml
```

There could be one or more conda environment files generated for each variant based on inputs 

> given to `build_env.py`. For example, if `build_env.py` is run without any build_type and python version 

Reviewer 2: ```suggestion
given to `build_env.py`. For example, if `build_env.py` is run without any `build_type` and python_versions 
```

> 3.7 and mpi_type as openmpi, then two files will be generated namely -

Reviewer 2: ```suggestion
`3.7` and mpi_type as `openmpi`, then two files will be generated namely -
```

> open-ce-conda-env-py3.7-cuda-openmpi.yaml, open-ce-conda-env-py3.7-cpu-openmpi.yaml. 

Reviewer 2: ```suggestion
`open-ce-conda-env-py3.7-cuda-openmpi.yaml`, `open-ce-conda-env-py3.7-cpu-openmpi.yaml`. 
```

> `build_env.py` can generate these environment files for a given Open-CE environment file 

Reviewer 2: ```suggestion
`build_env.py` can generate these target conda environment files for a given Open-CE environment file 
```

                        False)
==============================================================================
```

## Conda environment files

`build_env.py` also generates a conda environment file based on the configuration
 selected for build. For e.g. if `build_env.py` is run for tensorflow-env.yaml and
 for python version 3.7, build_type `cuda` and mpi_type being openmpi, then a
 conda environment file with name open-ce-conda-env-py3.7-cuda-openmpi.yaml gets
 generated. This environment file can be used to create a conda environment with
 the packages listed in tensorflow-env.yaml installed in it.

```shell
    ./open-ce/build_env.py --python_versions=3.7 --build_type=cuda --mpi_type=openmpi
    envs/tensorflow-env.yaml
```

>  Above command will output open-ce-conda-env-py3.7-cuda-openmpi.yaml in the specified

Reviewer: ```suggestion
 The above command will output open-ce-conda-env-py3.7-cuda-openmpi.yaml in the specified
```

 output folder (or by default `./condabuild` directory).

>  Following command can be used to create conda environment using the generated conda

Reviewer: ```suggestion
 The following command can be used to create a conda environment using the generated conda
```

 environment file -

```shell
    conda env create -f open-ce-conda-env-py3.7-cuda-openmpi.yaml
```

There could be one or more conda environment files generated for each variant based on inputs 
given to `build_env.py`. For example, if `build_env.py` is run without any build_type and python version 
3.7 and mpi_type as openmpi, then two files will be generated namely -
open-ce-conda-env-py3.7-cuda-openmpi.yaml, open-ce-conda-env-py3.7-cpu-openmpi.yaml. 

> `build_env.py` can generate these environment files for given Open-CE environment file 

Reviewer: ```suggestion
`build_env.py` can generate these environment files for a given Open-CE environment file 
```

> and provided build configuration even without performing actual build.

Reviewer: ```suggestion
and provided build configuration even without performing an actual build.
```
